xbmc/pvr/channels/PVRChannelGroup.cpp

```diff
@@ -376,63 +376,59 @@ void CPVRChannelGroup::SortByChannelNumber(void)
 
 CPVRChannelPtr CPVRChannelGroup::GetByClient(int iUniqueChannelId, int iClientID) const
 {
+  CPVRChannelPtr retval;
   CSingleLock lock(m_critSection);
 
-  for (unsigned int ptr = 0; ptr < m_members.size(); ptr++)
+  for (std::vector<PVRChannelGroupMember>::const_iterator it = m_members.begin(); !retval && it != m_members.end(); ++it)
   {
-    PVRChannelGroupMember groupMember = m_members.at(ptr);
-    if (groupMember.channel->UniqueID() == iUniqueChannelId &&
-        groupMember.channel->ClientID() == iClientID)
-      return groupMember.channel;
+    if ((*it).channel->UniqueID() == iUniqueChannelId &&
+        (*it).channel->ClientID() == iClientID)
+      retval = (*it).channel;
```

Comment:
These methods get called a lot, and they're only 3-4 lines, so I'd rather do these as optimised as possible. I'm going to change some of these to use a map rather than a vector and use the one that's used most (suppose that'll be channel number) as key, so at least one of these will be replaced by a `std::find()` call.